disdat/apply.py

```diff
@@ -285,19 +261,20 @@ def different_code_versions(code_version, lineage_obj):
     return False
 
     
-def resolve_bundle(pfs, pipe, is_left_edge_task, data_context):
+def resolve_bundle(pipe, data_context):
     """
     Args:
-        pfs: pipe fs object
         pipe: the pipe to investigate
-        is_left_edge_task: True if this task starts the DAG.
         data_context: the data context object from which we should resolve bundles.
 
     Returns:
         bool: True if bundle found (not re-running).  False if bundle not found or being regenerated
 
     """
 
+    from disdat.pipe import ExternalDepTask  # pipe.py->api.py->apply.py->pipe.ExternalDepTask fails b/c pipe importing
+    import disdat.api as api # 3.7 allows us to put this import at the top, but not 3.6.8
```

Comment:
what scary stuff are you doing here?  do you have a circular import chain?
seems very fragile

Reply:
Yes, I added a TODO comment to address this, however this ugly strategy works for everything 3.6.8 forward.